collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -534,6 +534,19 @@ public static void write(File file, byte[] data) {
         }
     }
 
+    /** Sorts file paths as if sorting the path components and extensions lexicographically. */
+    public static int comparePaths(String a, String b) {
+        // Regular string compareTo() is incorrect, because it will sort "/" and "."
+        // after other punctuation (e.g. "foo/bar" will sort AFTER "foo-2/bar" and
+        // "pic.jpg" will sort AFTER "pic-2.jpg").  Replacing these delimiters with
+        // '\u0000' and '\u0001' causes paths to sort correctly (assuming the paths
+        // don't already contain '\u0000' or '\u0001').  This is a bit of a hack,
+        // but it's a lot simpler and faster than comparing components one by one.
+        String aParts = a.replace('/', '\u0000').replace('.', '\u0001');
+        String bParts = b.replace('/', '\u0000').replace('.', '\u0001');
```

Comment:
The checkstyle regex is from the Google style guide and is there to avoid introducing type or usage prefixes (e.g. mFoo for a member, see https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml). Would you consider something like `sortingA` and `sortingB`?

Reply:
Ah, I see!